CIP-0155/README.md

```diff
@@ -0,0 +1,133 @@
+---
+CIP: 0155
+Title: SRV registry
+Status: Proposed
+Category: Network
+Authors:
+  - Marcin Szamotulski <marcin.szamotulski@iohk.io>
+Implementors: N/A
+Discussions: 
+  - [PR](https://github.com/cardano-foundation/CIPs/pull/1033)
+Created: 2025-04-22
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This CIP defines the procedure for assigning Service Record (SRV) prefixes for Cardano applications (like `cardano-node`, `mithril`, etc.).
+
+This means of sharing authoritative information between SPOs who deploy services and _decentralised protocol_ (DP) developers who write them.
+
+## Motivation: why is this CIP necessary?
+
+The **Cardano Ledger** allows the use of SRV records in the **SPO** registration certificate (see [register-stake-pool]).
+
+In this CIP, we propose to make them usable by decentralised applications running on `Cardano`.
+
+Having access to the ledger (either directly or through tools like `cardano-cli`) will allow decentralised protocols developers to construct networks based on registered relays. 
+
+All involved parties (SPOs, DP developers, etc) need to agree on SRV prefixes used by various DPs and thus a public registry governed by a public CIP process is required to foster DPs that co-exist with the **Cardano** network, like **Mithril** (see [CIP#137]).
+
+We consider DPs as applications, which construct their own network and relay on ledger peers published on the block-chain. This include node implementations, or 2nd-layer solutions like Mithril or Hydra.
+
+### Problem
+
+Bootstraping networks requires sharing information about available services.
+
+Historically, this was done by sharing IP or DNS names and PORT numbers, but this approach has its limitations.
+
+SRV records were designed to make deployment independent of hard-coded service end-points (see [RFC#2782]).
+
+Thus they can be instrumental in `Cardano`, since relays are registered on the blockchain through the registration certificate.
+
+By using SRV records in the registration certificate (which is supported by the `cardano-ledger`, but not by `cardano-node`), we wish to solve this problem not just for `Cardano` node implementation, but also for any DP that requires constructing its own network.
+
+SRV provides a mechanism for exposing DPs co-deployed with a node, like **mithril** or **hydra**.
+
+Making such services discoverable is one of the key features addressed by this CIP.
+
+In this CIP, we propose prefixes for both **Cardano** and **Mithril**; in the future, other services can be registered through a CIP process, thus starting a registry of prefixes used by the **Cardano** ecosystem.
+
+SPOs who deploy services need to configure their system according to the registry, e.g. SPO's cardano relays node MUST be available at `_cardano._tcp.<SPO_DOMAIN_NAME>`, as other nodes on the system will be looking at this address.
```

Comment:
It sounds like this sentence is describing the *current* state, correct? i.e. already the SRV records used by SPOs are expected to / required to begin with `_cardano._tcp`; and this proposal just changes that from a singular value to a "pre"fix, so it can be extended for other services?

Or is this intended to be a breaking change, where there was no enforcement on the SRV record format so far, and this invalidates those, requiring them to be more consistent?

Reply:
If so, and there's not already a standard / requirement of using the `_cardano._tcp` prefix, then this represents a much more disruptive change for SPOs, and should be addressed in the implementation plan.

Reply:
> It sounds like this sentence is describing the current state, correct? i

If one has an SRV record that uses a different semantics than what this CIP proposes, it will be broken. However, this doesn't affect `cardano-node`, as it's not being used. So from the `cardano-node` perspective: what used to be broken remains broken :grin:.